crmsh/bootstrap.py

```diff
@@ -1342,60 +1293,13 @@ def init_qdevice():
             utils.fatal("Failed to copy ssh key: {}".format(err))
     # Start qdevice service if qdevice already configured
     if utils.is_qdevice_configured() and not confirm("Qdevice is already configured - overwrite?"):
-        start_qdevice_service()
+        qdevice_inst.start_qdevice_service()
         return
-
+    qdevice_inst.set_cluster_name()
     # Validate qnetd node
     qdevice_inst.valid_qnetd()
```

Comment:
Ideally, calling qdevice_inst.valid_qnetd() here without locking will have the potential race condition, just in case there are multiple clusters with the same name. It should be protected by locking, hence better move it into  config_and_start_qdevice().

Reply:
Is that harmful if have race condition here for the `valid_qnetd` function?

Reply:
First of all, this is a typical scenario to put "compare and change" logic under the locking context.

Now, given two fresh clusters with the same name, they will likely  go over config_and_start_qdevice() twice. One overwrite another.

I agree it is a very rare corner case, and not likely happen. But, it is better to make the code fits well with the theory, that will help code maintenance.

Reply:
```
    @qnetd_lock_for_same_cluster_name
    def config_and_start_qdevice(self):
```
This function was locked by `qnetd_lock_for_same_cluster_name` before enter `config_and_start_qdevice`, wouldn't go over twice

Reply:
aha, you are right. Then, no race condition! That's a little tricky here since there are two different locks. Good for me now!